modules/openapi-generator/src/main/resources/kotlin-spring/apiInterface.mustache

```diff
@@ -77,13 +71,13 @@ interface {{classname}} {
         authorizations = [{{#authMethods}}Authorization(value = "{{name}}"{{#isOAuth}}, scopes = [{{#scopes}}AuthorizationScope(scope = "{{scope}}", description = "{{description}}"){{#hasMore}}, {{/hasMore}}{{/scopes}}]{{/isOAuth}}){{#hasMore}}, {{/hasMore}}{{/authMethods}}]{{/hasAuthMethods}})
     @ApiResponses(
         value = [{{#responses}}ApiResponse(code = {{{code}}}, message = "{{{message}}}"{{#baseType}}, response = {{{baseType}}}::class{{/baseType}}{{#containerType}}, responseContainer = "{{{containerType}}}"{{/containerType}}){{#hasMore}},{{/hasMore}}{{/responses}}]){{/swaggerAnnotations}}
-    @RequestMapping(
-            value = ["{{#lambda.escapeDoubleQuote}}{{path}}{{/lambda.escapeDoubleQuote}}"],{{#singleContentTypes}}{{#hasProduces}}
-            produces = "{{{vendorExtensions.x-accepts}}}",{{/hasProduces}}{{#hasConsumes}}
-            consumes = "{{{vendorExtensions.x-contentType}}}",{{/hasConsumes}}{{/singleContentTypes}}{{^singleContentTypes}}{{#hasProduces}}
-            produces = [{{#produces}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/produces}}], {{/hasProduces}}{{#hasConsumes}}
-            consumes = [{{#consumes}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/consumes}}],{{/hasConsumes}}{{/singleContentTypes}}
-            method = [RequestMethod.{{httpMethod}}])
+    @{{#lambda.titlecase}}{{#lambda.lowercase}}{{httpMethod}}{{/lambda.lowercase}}{{/lambda.titlecase}}Mapping(
```

Comment:
Using `{{#lambda.titlecase}}{{#lambda.lowercase}} ... {{/lambda.lowercase}}{{/lambda.titlecase}}` is fine. Later I may switch it to something else (e.g. x-kotlin-http-method) to make it easier for new users to understand.

Reply:
Hello,

I just encountered a case where this change is actually blocking for me to upgrade from 4.3.1 to 5.x.x.

I do have an HTTP OPTIONS in my OpenAPI. This change generates an `@OptionsMapping`, but it does not exist in Spring. The previous version with `@RequestMapping` and setting the `RequestMethod` was working for all HTTP methods. I admit, it is more verbose, but at least it covers the various cases.

This change is only viable once there are annotations `*Mapping` for every `RequestMethod` values in Spring.

You can find more information in this Spring issue where it is a bit discussed: https://github.com/spring-projects/spring-framework/issues/18565

Can this change be reverted?